server/src/main/java/com/cloud/template/TemplateManagerImpl.java

```diff
@@ -2198,6 +2202,10 @@ private VMTemplateVO updateTemplateOrIso(BaseUpdateTemplateOrIsoCmd cmd) {
             template.setDynamicallyScalable(isDynamicallyScalable);
         }
 
+        if (templateType != null && templateType == TemplateType.SYSTEM) {
+            copyTemplateToAllZones(id);
+        }
```

Comment:
@davidjumani should the template be changed to cross-zone when it is done ?

Reply:
Makes sense. Added the change